src/main/java/com/internet/shop/dao/impl/ShoppingCartDaoListImpl.java

```diff
@@ -0,0 +1,56 @@
+package com.internet.shop.dao.impl;
+
+import com.internet.shop.dao.ShoppingCartDao;
+import com.internet.shop.db.Storage;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.model.Product;
+import com.internet.shop.model.ShoppingCart;
+import java.util.Optional;
+import java.util.stream.IntStream;
+
+@Dao
+public class ShoppingCartDaoListImpl implements ShoppingCartDao {
+    @Override
+    public ShoppingCart create(ShoppingCart shoppingCart) {
+        Storage.addShoppingCart(shoppingCart);
+        return shoppingCart;
+    }
+
+    @Override
+    public ShoppingCart addProduct(ShoppingCart shoppingCart, Product product) {
+        shoppingCart.getProducts().add(product);
+        return shoppingCart;
+    }
+
+    @Override
+    public boolean deleteProduct(ShoppingCart shoppingCart, Product product) {
+        return shoppingCart.getProducts().remove(product);
+    }
+
+    @Override
+    public void clear(ShoppingCart shoppingCart) {
+        shoppingCart.getProducts().clear();
+    }
```

Comment:
Also to the service layer. 
Dao should contain CRUD operations only - direct access to the DataBase.